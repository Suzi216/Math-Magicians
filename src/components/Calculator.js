import React, { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [items, setStates] = useState({
    total: 0,
    next: null,
    operation: null,
  });
  const handleCalc = (e) => {
    const data = e.target.value;
    const item = calculate(items, data);
    setStates(item);
  };

  const { total, next, operation } = items;
  return (
    <div className="container">
      <table cellSpacing="0">
        <tr>
          <td colSpan="3">
            <div className="result">
              {total}
              {operation}
              {next}
            </div>
          </td>
        </tr>
        <tr>
          <td>
            <input className="others" type="button" value="AC" onClick={handleCalc} />
            {' '}
          </td>
          <td>
            <input className="others" type="button" value="+/-" onClick={handleCalc} />
            {' '}
          </td>
          <td>
            <input className="others" type="button" value="%" onClick={handleCalc} />
            {' '}
          </td>
          <td>
            <input className="operants" type="button" value="รท" onClick={handleCalc} />
            {' '}
          </td>
        </tr>
        <tr>
          <td>
            <input className="others" type="button" value="7" onClick={handleCalc} />
            {' '}
          </td>
          <td>
            <input className="others" type="button" value="8" onClick={handleCalc} />
            {' '}
          </td>
          <td>
            <input className="others" type="button" value="9" onClick={handleCalc} />
            {' '}
          </td>
          <td>
            <input className="operants" type="button" value="x" onClick={handleCalc} />
            {' '}
          </td>
        </tr>
        <tr>
          <td>
            <input className="others" type="button" value="4" onClick={handleCalc} />
            {' '}
          </td>
          <td>
            <input className="others" type="button" value="5" onClick={handleCalc} />
            {' '}
          </td>
          <td>
            <input className="others" type="button" value="6" onClick={handleCalc} />
            {' '}
          </td>
          <td>
            <input className="operants" type="button" value="-" onClick={handleCalc} />
            {' '}
          </td>
        </tr>
        <tr>
          <td>
            <input className="others" type="button" value="1" onClick={handleCalc} />
            {' '}
          </td>
          <td>
            <input className="others" type="button" value="2" onClick={handleCalc} />
            {' '}
          </td>
          <td>
            <input className="others" type="button" value="3" onClick={handleCalc} />
            {' '}
          </td>
          <td>
            <input className="operants" type="button" value="+" onClick={handleCalc} />
            {' '}
          </td>
        </tr>
        <tr>
          <td colSpan="2">
            <input className="others" type="button" value="0" onClick={handleCalc} />
            {' '}
          </td>
          <td>
            <input className="others" type="button" value="." onClick={handleCalc} />
            {' '}
          </td>
          <td>
            <input className="operants" type="button" value="=" onClick={handleCalc} />
            {' '}
          </td>
        </tr>
      </table>
    </div>

  );
};

export default Calculator;
