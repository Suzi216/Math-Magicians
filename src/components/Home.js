import React from 'react';
import '../styles/home.css';

const Home = () => (
  <div className="home-container">
    <h3>What is a calculator</h3>
    <p>
      {' '}
      A calculator is a device that performs arithmetic operations on numbers.
      Basic calculators can do only addition, subtraction, multiplication
      and division mathematical calculations.However, more sophisticated
      calculators can handle exponential operations,square roots, logarithms,
      trigonometric functions and hyperbolic functions.
    </p>
    <p>
      Most calculators these days require electricity to operate or are
      battery-powered calculators. Calculators work by performing programmed
      functions based on numerical inputs. Before the electronic calculator
      (circa 1970), a more primitive calculator, the slide rule,
      was commonly used. It consisted of a slat of wood called the slide that
      could be moved in and out of a reinforced pair of slats. Both the slide and
      the outer pair of slats had calibrated numerical scales.
    </p>
    <h3>Types of calculators</h3>
    <h5>Basic calculators</h5>
    <p>
      The most basic calculator is the four-function calculator, which can perform
      basic arithmetic such as addition, subtraction, multiplication and division.
    </p>
    <h5>Scientific calculators</h5>
    <p>
      This type of calculator usually has more buttons than a standard calculator,
      as it needs to be able to perform trigonometric functions, logarithms,
      sine/cosine and exponential operations. The scientific calculator also
      usually has a larger display to view long equations and see more digits
      at a time.
    </p>
    <h5>Graphing calculators</h5>
    <p>
      The graphing calculator also can graph equations, for example,
      with more advanced mathematics like trigonometry, on a coordinate plane.
    </p>
  </div>
);
export default Home;
