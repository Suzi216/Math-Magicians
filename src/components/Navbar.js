import React from 'react';
import { NavLink } from 'react-router-dom';
import '../styles/navbar.css';

const Navbar = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Home',
    },
    {
      id: 2,
      path: '/quotes',
      text: 'Quotes',
    },
    {
      id: 3,
      path: '/calculator',
      text: 'Calculator',
    },
  ];
  return (
    <nav>
      <ul className="navBar">
        <h1>Math Magicians</h1>
        {links.map((link) => <li key={link.id}><NavLink to={link.path}>{link.text}</NavLink></li>)}
      </ul>
    </nav>
  );
};
export default Navbar;
