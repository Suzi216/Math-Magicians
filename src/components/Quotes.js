import React from 'react';
import '../styles/homeANDquotes.css';

const Quotes = () => (
  <div className="quote-container">
    <q>
      Spirit of Math -  Mathematics is not about numbers, equations,
      computations or algorithms; it is about understanding.
      - Willian Paul Thurston
    </q>
  </div>
);
export default Quotes;
