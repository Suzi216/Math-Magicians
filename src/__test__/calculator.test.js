import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator'
import Home from '../components/Home';
import Navbar from '../components/Navbar';
import Quotes from '../components/Quotes';

it('renders correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
