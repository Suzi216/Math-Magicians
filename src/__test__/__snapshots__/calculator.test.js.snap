// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <h2>
     Lets do some math!
  </h2>
  <table
    cellSpacing="0"
  >
    <tr>
      <td
        colSpan="3"
      >
        <div
          className="result"
        >
          0
        </div>
      </td>
    </tr>
    <tr>
      <td>
        <input
          className="others"
          onClick={[Function]}
          type="button"
          value="AC"
        />
         
      </td>
      <td>
        <input
          className="others"
          onClick={[Function]}
          type="button"
          value="+/-"
        />
         
      </td>
      <td>
        <input
          className="others"
          onClick={[Function]}
          type="button"
          value="%"
        />
         
      </td>
      <td>
        <input
          className="operants"
          onClick={[Function]}
          type="button"
          value="รท"
        />
         
      </td>
    </tr>
    <tr>
      <td>
        <input
          className="others"
          onClick={[Function]}
          type="button"
          value="7"
        />
         
      </td>
      <td>
        <input
          className="others"
          onClick={[Function]}
          type="button"
          value="8"
        />
         
      </td>
      <td>
        <input
          className="others"
          onClick={[Function]}
          type="button"
          value="9"
        />
         
      </td>
      <td>
        <input
          className="operants"
          onClick={[Function]}
          type="button"
          value="x"
        />
         
      </td>
    </tr>
    <tr>
      <td>
        <input
          className="others"
          onClick={[Function]}
          type="button"
          value="4"
        />
         
      </td>
      <td>
        <input
          className="others"
          onClick={[Function]}
          type="button"
          value="5"
        />
         
      </td>
      <td>
        <input
          className="others"
          onClick={[Function]}
          type="button"
          value="6"
        />
         
      </td>
      <td>
        <input
          className="operants"
          onClick={[Function]}
          type="button"
          value="-"
        />
         
      </td>
    </tr>
    <tr>
      <td>
        <input
          className="others"
          onClick={[Function]}
          type="button"
          value="1"
        />
         
      </td>
      <td>
        <input
          className="others"
          onClick={[Function]}
          type="button"
          value="2"
        />
         
      </td>
      <td>
        <input
          className="others"
          onClick={[Function]}
          type="button"
          value="3"
        />
         
      </td>
      <td>
        <input
          className="operants"
          onClick={[Function]}
          type="button"
          value="+"
        />
         
      </td>
    </tr>
    <tr>
      <td
        colSpan="2"
      >
        <input
          className="others"
          onClick={[Function]}
          type="button"
          value="0"
        />
         
      </td>
      <td>
        <input
          className="others"
          onClick={[Function]}
          type="button"
          value="."
        />
         
      </td>
      <td>
        <input
          className="operants"
          onClick={[Function]}
          type="button"
          value="="
        />
         
      </td>
    </tr>
  </table>
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="home-container"
>
  <h3>
    What is a calculator
  </h3>
  <p>
     
    A calculator is a device that performs arithmetic operations on numbers. Basic calculators can do only addition, subtraction, multiplication and division mathematical calculations.However, more sophisticated calculators can handle exponential operations,square roots, logarithms, trigonometric functions and hyperbolic functions.
  </p>
  <p>
    Most calculators these days require electricity to operate or are battery-powered calculators. Calculators work by performing programmed functions based on numerical inputs. Before the electronic calculator (circa 1970), a more primitive calculator, the slide rule, was commonly used. It consisted of a slat of wood called the slide that could be moved in and out of a reinforced pair of slats. Both the slide and the outer pair of slats had calibrated numerical scales.
  </p>
  <h3>
    Types of calculators
  </h3>
  <h5>
    Basic calculators
  </h5>
  <p>
    The most basic calculator is the four-function calculator, which can perform basic arithmetic such as addition, subtraction, multiplication and division.
  </p>
  <h5>
    Scientific calculators
  </h5>
  <p>
    This type of calculator usually has more buttons than a standard calculator, as it needs to be able to perform trigonometric functions, logarithms, sine/cosine and exponential operations. The scientific calculator also usually has a larger display to view long equations and see more digits at a time.
  </p>
  <h5>
    Graphing calculators
  </h5>
  <p>
    The graphing calculator also can graph equations, for example, with more advanced mathematics like trigonometry, on a coordinate plane.
  </p>
</div>
`;
